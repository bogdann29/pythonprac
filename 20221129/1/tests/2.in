class C(metaclass=dump):
    def __init__(self, val):
        self.val = val

    def add(self, other, another=None):
        return self.val + other + (another or self.val)

    def f(self, a=5, b=6):
        return (a * b) + self.val

c = C(111)
print(c.add(9))
print(c.add(9,1))
print(c.f())
print(c.f(b=2))