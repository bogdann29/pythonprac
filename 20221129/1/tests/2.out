__init__: (111,), {}
add: (9,), {}
231
add: (9, 1), {}
121
f: (), {}
141
f: (), {'b': 2}
121
